src/lib/one-graph/cli-client.js

```diff
@@ -273,6 +274,33 @@ const regenerateFunctionsFileFromOperationsFile = (input) => {
   generateFunctionsFile({ netlifyGraphConfig, schema, operationsDoc: appOperationsDoc, functions, fragments })
 }
 
+/**
+ * Lockfile Operations
+ */
+
+/**
+ * Persist the Netlify Graph lockfile on disk
+ * @param {object} input
+ * @param {string} input.siteRoot The GraphQL schema to use when generating code
+ * @param {NetlifyGraphLockfile.V0_format} input.lockfile
+ */
+const writeLockfile = ({ lockfile, siteRoot }) => {
+  writeFileSync(path.join(siteRoot, NetlifyGraphLockfile.defaultLockFileName), JSON.stringify(lockfile, null, 2))
```

Comment:
I suspect in the near future we may want a `normalizeLockfile` function that inserts all the known keys in a deterministic order so that the JSON file is always written out in the same order as well, to prevent spurious changes in git

Reply:
I like that we enforce that invariant in the future. It should hopefully be the case right now if the entrypoint is always `NetlifyGraphLockfile.createLockfile`